common/src/main/java/com/wynntils/features/utilities/InputTranscriptionFeature.java

```diff
@@ -0,0 +1,287 @@
+        }
+    }
+
+    @SubscribeEvent
+    public void onChatScreenKeyTyped(ChatScreenKeyTypedEvent event) {
+        if (event.getKeyCode() != GLFW.GLFW_KEY_BACKSPACE) return;
+        if (!(McUtils.mc().screen instanceof ChatScreen chatScreen)) return;
+        if (!chatScreen.input.getHighlighted().isBlank()) return;
+
+        String beforeCursor = chatScreen.input.getValue().substring(0, chatScreen.input.getCursorPosition());
+        String afterCursor = chatScreen.input.getValue().substring(chatScreen.input.getCursorPosition());
+
+        if (beforeCursor.isBlank()
+                || !Models.WynnAlphabet.getWynnicNumbers().contains(beforeCursor.charAt(beforeCursor.length() - 1)))
+            return;
+
+        String currentNumsStr = getWynnicNumBeforeCursor(beforeCursor);
+
+        int wynnicNum = Models.WynnAlphabet.wynnicNumToInt(currentNumsStr);
+
+        String backspacedNum = String.valueOf(wynnicNum);
+
+        if (backspacedNum.length() > 1) {
+            event.setCanceled(true);
+
+            backspacedNum = backspacedNum.substring(0, backspacedNum.length() - 1);
+
+            int index = beforeCursor.lastIndexOf(currentNumsStr);
+
+            beforeCursor = beforeCursor.substring(0, index);
+
+            updateInput(beforeCursor, afterCursor, Integer.parseInt(backspacedNum), chatScreen.input);
+        }
+    }
+
+    @SubscribeEvent
+    public void onChatSend(ChatSentEvent event) {
+        if (event.getMessage().isBlank()) return;
+
+        String message = event.getMessage();
+
+        if (useBrackets.get() && containsBrackets(message)) {
+            String updatedMessage = transcriptSentMessage(message, event);
+
+            updatedMessage = updatedMessage.substring(0, Math.min(updatedMessage.length(), MAX_CHAT_LENGTH));
+
+            McUtils.mc().getConnection().sendChat(updatedMessage);
+        } else {
+            Models.WynnAlphabet.setSelectedAlphabet(WynnAlphabet.DEFAULT);
+        }
+    }
+
+    @SubscribeEvent
+    public void onCommandSent(CommandSentEvent event) {
+        String command = event.getCommand();
+
+        if (useBrackets.get() && containsBrackets(command)) {
+            String updatedCommand = transcriptSentMessage(command, event);
+
+            updatedCommand = updatedCommand.substring(0, Math.min(updatedCommand.length(), MAX_CHAT_LENGTH));
+
+            McUtils.sendCommand(updatedCommand);
+        } else {
+            Models.WynnAlphabet.setSelectedAlphabet(WynnAlphabet.DEFAULT);
+        }
+    }
+
+    private void addAlphabetButton(ChatScreen chatScreen, int xOffset, WynnAlphabet alphabet) {
+        chatScreen.addRenderableWidget(new WynnAlphabetButton(xOffset, chatScreen.height - 14, 12, 12, alphabet));
+    }
+
+    private boolean containsBrackets(String message) {
+        return (message.contains("{") && message.contains("}")) || (message.contains("<") && message.contains(">"));
```

Comment:
I am worried this is too specific and might be surprising. What about this syntax instead?
`{{wynnic text}} and <<gavellian>>`

It is not much harder to type, but much less likely to be typed by mistake.

Also, for `{}`, we should think about how that plays with the function parsing. It would be nice if we could insert wynnic text in info boxes (for the real hardcore players! :-)) without breaking the function parsing. 

So possibly use `[[ wynnic ]]` instead, to avoid that completely. Or check that the function string parser does not choke on `{{` but can pass it on unaltered.

Reply:
I've not actually looked into functions so might take me a bit to figure out, but I'll certainly give it a try

Reply:
I've put it in here, is this the best place for it?

```
public StyledText[] doFormatLines(String templateString) {
        StringBuilder resultBuilder = new StringBuilder();

        if (Models.WynnAlphabet.containsBrackets(templateString)) {
            templateString = Models.WynnAlphabet.transcribeBracketedText(templateString);
        }
```

![image](https://github.com/Wynntils/Artemis/assets/8337467/e1c56f84-18f8-4c5d-a1d7-25875bd9fda6)

Reply:
That is a much better idea (: